lib/pbench/test/unit/server/test_requests.py

```diff
@@ -188,6 +189,32 @@ def test_bad_length_header_upload(
         self.verify_logs(caplog)
         assert not self.cachemanager_created
 
+    def test_bad_metadata_upload(
+        self, client, caplog, server_config, setup_ctrl, pbench_drb_token
+    ):
+        with freeze_time("1970-01-01 00:42:00"):
+            response = client.put(
+                self.gen_uri(server_config),
+                headers={
+                    "Authorization": "Bearer " + pbench_drb_token,
+                    "controller": self.controller,
+                    "Content-MD5": "ANYMD5",
+                    "Content-Length": "STRING",
+                },
+                query_string={
+                    "metadata": "foobar.badpath:data,server.deletion:3000-12-25T23:59:59+00:00"
+                },
+            )
+        assert response.status_code == HTTPStatus.BAD_REQUEST
+        json = response.json
+        assert "errors" in json and "message" in json
+        assert json["message"] == "at least one specified metadata key is invalid"
+        assert json["errors"] == [
+            "Key foobar.badpath is invalid or isn't settable",
+            "Metadata key 'server.deletion' value '3000-12-25T23:59:59+00:00' for dataset must be a date/time before 1979-12-30",
```

Comment:
This is an interesting complaint...the deletion value must be **_before_** the year 1980??

Reply:
Based on the upload timestamp, yeah.